src/Tool-CriticBrowser/CBSelectPackageBrowser.class.st

```diff
@@ -0,0 +1,174 @@
+
+{ #category : #specs }
+CBSelectPackageBrowser class >> defaultSpec [
+	^ SpBoxLayout newVertical
+		add:
+			(SpPanedLayout newHorizontal
+				add: #toSelectPresenter;
+				add: #selectedPresenter;
+				yourself);
+		add:
+			(SpBoxLayout newHorizontal
+				add: #removeTestPackageButton;
+				add: #addTestCaseClassesCheckBox;
+				yourself)
+			withConstraints: [ :contraint | contraint height: 25 ];
+		add:
+			(SpBoxLayout newHorizontal
+				add: #addButton;
+				add: #addAllButton;
+				add: #removeButton;
+				add: #removeAllButton;
+				add: #nextButton;
+				yourself)
+			withConstraints: [ :contraint | contraint height: 25 ];
+		yourself
+]
+
+{ #category : #'instance creation' }
+CBSelectPackageBrowser class >> open [
+	<script>
+	self new
+		items: RPackageOrganizer default packages;
+		openWithSpec
+]
+
+{ #category : #private }
+CBSelectPackageBrowser >> addAllItems [
+	selectedPresenter items: (self sort: selectedPresenter items withAll: toSelectPresenter items).
+	toSelectPresenter items: #()
+]
+
+{ #category : #private }
+CBSelectPackageBrowser >> addItems [
+	toSelectPresenter selectedItems
+		ifNotNil: [ :items | 
+			selectedPresenter items: (self sort: selectedPresenter items withAll: items).
+			toSelectPresenter items: (self sort: toSelectPresenter items withoutAll: items) ].
+	toSelectPresenter unselectAll
+]
+
+{ #category : #accessing }
+CBSelectPackageBrowser >> addTestCaseClasses [
+	^ addTestCaseClassesCheckBox
+
+]
+
+{ #category : #initialization }
+CBSelectPackageBrowser >> initialize [
+	super initialize.
+	testPackagesToRemove := OrderedCollection new
+]
+
+{ #category : #initialization }
+CBSelectPackageBrowser >> initializeWidgets [
+	super initializeWidgets.
+	addTestCaseClassesCheckBox := self newCheckBox.
+	removeTestPackageButton := self newButton.
+	toSelectPresenter := self newList.
+	selectedPresenter := self newList.
+	
+	toSelectPresenter displayBlock: [ :package | package packageName ].
+	selectedPresenter displayBlock: [ :package | package packageName ].
+	selectedPresenter beMultipleSelection.
+	toSelectPresenter beMultipleSelection.
+	self setRemoveTestPackageButton.
+	addTestCaseClassesCheckBox
+		label: 'Remove TestCases';
+		state: true
+]
+
+{ #category : #initialization }
+CBSelectPackageBrowser >> initializeWindow: aWindowPresenter [
+	super initializeWindow: aWindowPresenter.
+	aWindowPresenter title: 'Select packages'
+]
+
+{ #category : #api }
+CBSelectPackageBrowser >> nextAction [
+	CBSelectRuleBrowser openWithEnvironment: (RBPackageEnvironment new packages: selectedPresenter items) removeTestCase: self addTestCaseClasses state.
+	self withWindowDo: #close
+]
+
+{ #category : #private }
+CBSelectPackageBrowser >> removeAllItems [
+	toSelectPresenter items: (self sort: toSelectPresenter items withAll: selectedPresenter items).
+	selectedPresenter items: #()
+]
+
+{ #category : #private }
+CBSelectPackageBrowser >> removeItems [
+	selectedPresenter selectedItems
+		ifNotNil: [ :items | 
```

Comment:
Isn't `selectedPresenter` a list? Then selectedItems should never be nil (it's either a collection with elements or empty), then the if is not necessary, is it?